scripts/system/create/entitySelectionTool/entitySelectionTool.js

```diff
@@ -31,6 +32,46 @@ function deepCopy(v) {
     return JSON.parse(JSON.stringify(v));
 }
 
+function getDuplicateAppendedName(name) {
+    var appendiceChar = "(0123456789)";
+    var currentChar = "";
+    var rippedName = name;
+    var existingSequence = 0;
+    var sequenceReader = "";
+    for (var i = name.length - 1; i >= 0; i--) {
+        currentChar = name.charAt(i);
+        if (i === (name.length - 1) && currentChar !== ")") {
+            rippedName = name;
+            break;
+        } else {
+            if (appendiceChar.indexOf(currentChar) === -1) {
+                rippedName = name;
+                break;
+            } else {
+                if (currentChar == "(" && i === name.length - 2) {
+                    rippedName = name;
+                    break;
+                } else {
+                    if (currentChar == "(") {
+                        rippedName = name.substr(0, i-1);
+                        existingSequence = parseInt(sequenceReader);
+                        break;
+                    } else {
+                        sequenceReader = currentChar + sequenceReader;
+                    }
+                }
+            }
+        }
+    }
```

Comment:
this logic *seems* sound to me, but it's a little hard to follow, which means it will also be a little hard to debug if something ever goes wrong.

I wonder if it would be easier to use regexs?  I don't know what JS regex support looks like

Reply:
I've simply used something I trust... regex is just not intuitive "for me" and I don't know what surprise it could bring.
I will check that option.

Reply:
yeah not super important, just wanted to mention it in case it would be easier